src/ui/public/field_editor/field_editor.js

```diff
@@ -509,19 +559,39 @@ export class FieldEditor extends PureComponent {
   render() {
     const { isReady, isCreating, scriptingLangs, field, showScriptingHelp } = this.state;
 
+    let scriptDisabledCallout;
+    let scriptWarningCallout;
+    let scriptHelpFlyout;
+    if (field.scripted) {
+      scriptDisabledCallout = (
+        <ScriptingDisabledCallOut isVisible={!scriptingLangs.length} />
+      );
+      scriptWarningCallout = (
+        <ScriptingWarningCallOut isVisible />
+      );
+      scriptHelpFlyout = (
+        <ScriptingHelpFlyout
+          isVisible={showScriptingHelp}
+          onClose={this.hideScriptingHelp}
+          indexPattern={this.props.indexPattern}
+          lang={field.lang}
+          name={field.name}
+          script={field.script}
+          executeScript={executeScript}
+        />
+      );
+    }
+
     return isReady ? (
       <div>
         <EuiText>
           <h3>{isCreating ? 'Create scripted field' : `Edit ${field.name}`}</h3>
         </EuiText>
         <EuiSpacer size="m" />
         <EuiForm>
-          <ScriptingDisabledCallOut isVisible={field.scripted && !scriptingLangs.length} />
-          <ScriptingWarningCallOut isVisible={field.scripted} />
-          <ScriptingHelpFlyout
-            isVisible={field.scripted && showScriptingHelp}
-            onClose={this.hideScriptingHelp}
-          />
+          {scriptDisabledCallout}
```

Comment:
I'd love to see this moved out to a separate `renderScriptingPanels()` method or similar, which can then check `field.scripted` and return these components, or null. then this render method can avoid the `let` vars and use `{this.renderScriptingPanels()}`